frame/contracts/src/lib.rs

```diff
@@ -618,89 +372,294 @@ decl_module! {
+		/// contract has enough balance to pay for its rent.
+		ContractNotEvictable,
+		/// A storage modification exhausted the 32bit type that holds the storage size.
 		///
-		/// If contract is not evicted as a result of this call, [`Error::ContractNotEvictable`]
-		/// is returned and the sender is not eligible for the reward.
-		#[weight = T::WeightInfo::claim_surcharge(T::MaxCodeSize::get() / 1024)]
-		pub fn claim_surcharge(
-			origin,
-			dest: T::AccountId,
-			aux_sender: Option<T::AccountId>
-		) -> DispatchResultWithPostInfo {
-			let origin = origin.into();
-			let (signed, rewarded) = match (origin, aux_sender) {
-				(Ok(frame_system::RawOrigin::Signed(account)), None) => {
-					(true, account)
-				},
-				(Ok(frame_system::RawOrigin::None), Some(aux_sender)) => {
-					(false, aux_sender)
-				},
-				_ => Err(Error::<T>::InvalidSurchargeClaim)?,
-			};
+		/// This can either happen when the accumulated storage in bytes is too large or
+		/// when number of storage items is too large.
+		StorageExhausted,
+		/// A contract with the same AccountId already exists.
+		DuplicateContract,
+	}
 
-			// Add some advantage for block producers (who send unsigned extrinsics) by
-			// adding a handicap: for signed extrinsics we use a slightly older block number
-			// for the eviction check. This can be viewed as if we pushed regular users back in past.
-			let handicap = if signed {
-				T::SignedClaimHandicap::get()
-			} else {
-				Zero::zero()
-			};
+	#[pallet::type_value]
+	pub fn Null() -> u64 {
+		0
+	}
 
-			// If poking the contract has lead to eviction of the contract, give out the rewards.
-			match Rent::<T, PrefabWasmModule<T>>::try_eviction(&dest, handicap)? {
-				(Some(rent_payed), code_len) => {
-					T::Currency::deposit_into_existing(
-						&rewarded,
-						T::SurchargeReward::get().min(rent_payed),
-					)
-					.map(|_| PostDispatchInfo {
-						actual_weight: Some(T::WeightInfo::claim_surcharge(code_len / 1024)),
-						pays_fee: Pays::No,
-					})
-					.map_err(Into::into)
-				}
-				(None, code_len) => Err(Error::<T>::ContractNotEvictable.with_weight(
-					T::WeightInfo::claim_surcharge(code_len / 1024)
-				)),
+	/// Current cost schedule for contracts.
+	#[pallet::storage]
+	#[pallet::getter(fn current_schedule)]
+	pub(super) type CurrentSchedule<T: Config> = StorageValue<_, Schedule<T>, ValueQuery>;
+
+	/// A mapping from an original code hash to the original code, untouched by instrumentation.
+	#[pallet::storage]
+	pub type PristineCode<T: Config> = StorageMap<_, Identity, CodeHash<T>, Vec<u8>>;
+
+	/// A mapping between an original code hash and instrumented wasm code, ready for execution.
+	#[pallet::storage]
+	pub type CodeStorage<T: Config> = StorageMap<_, Identity, CodeHash<T>, PrefabWasmModule<T>>;
+
+	/// The subtrie counter.
+	#[pallet::storage]
+	pub type AccountCounter<T: Config> = StorageValue<_, u64, ValueQuery, Null>;
```

Comment:
The default value for u64 is 0 thus I think it is better to use the default value directly.
But if you prefer to explicit it you can close this suggestion.

```suggestion
	pub type AccountCounter<T: Config> = StorageValue<_, u64, ValueQuery>;
```

Reply:
Ahh okay I did not know that you can just leave out the last parameter so that it picks the default. I thought leaving it out just meant "no default value" and it requires you to set it as a genesis value.